core/src/main/java/org/testcontainers/containers/Container.java

```diff
@@ -141,6 +141,18 @@ default void addFileSystemBind(final String hostPath, final String containerPath
      */
     SELF waitingFor(@NonNull WaitStrategy waitStrategy);
 
+    /**
+     * Configure the container using the provided consumer.
+     *
+     * @param consumer the consumer of the container
+     * @return this
+     */
+    default SELF with(Consumer<SELF> consumer)
```

Comment:
While `with` is indeed one of the most logical choices for the name of such method, I am thinking about Groovy/Kotlin/Scala and others where it already has a meaning and may even change the source compatibility.

Could you please find another name that will not overlap with existing extension methods from popular JVM languages (`apply`, `with`, `do`, `tap`, `let`, `also`, `run`)?

Reply:
That makes sense. I changed it to `update`, but open to suggestions. I also thought about `withConsumer` but seems a bit implementation-specific and is confusing due to `withLogConsumer`.

Reply:
I would suggest keeping a `with` prefixed method name for easier discoverability. 
Maybe `withApplied()` ? Or `withAppliedConsumer()`?

I was also thinking about `withConfiguration()`, but maybe too unspecific?

Reply:
crazy idea.... `tune`? :D

Reply:
Does make sense from a semantic standpoint, but I wonder about discoverability (but reads really cool, tune the container to your testing needs...). 

`withTuning()`? ^^

Reply:
there are many other `with*` methods for setting container properties, maybe `applyTuning`? :D e ron don don 😂 

cc @rnorth